src/neubot/negotiate_impl.hpp

```diff
@@ -127,16 +134,31 @@ static inline void loop_req_negotiate(Var<Transport> transport, Callback<Error>
 static inline void run_impl(Callback<Error> cb, Settings settings,
                             Var<Reactor> reactor, Var<Logger> logger) {
 
-    if (settings.at("url") == "") {
-        settings["http/url"] = "http://127.0.0.1";
+
+    std::string tool = "neubot";
+    std::string url;
+
+    if (settings["url"] == "") {
+        mlabns::query(
+           tool, [&url](Error error, mlabns::Reply reply) {
+               if (error) {
+                   logger -> warn("Error: %d", (int) error);
+                   cb(error);
+                   return;
+               }
+               url = reply.url;
+           });
+        settings["http/url"] = url;
```

Comment:
@bassosimone I have taken this part of code from the example of use of mlabns, but I don't understand why it doesn't work. It seems like it doesn't return an URL.